core/src/main/java/org/mule/TransformationService.java

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule;
+
+import static org.mule.api.transport.PropertyScope.INBOUND;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.MuleMessage;
+import org.mule.api.MuleRuntimeException;
+import org.mule.api.transformer.Converter;
+import org.mule.api.transformer.DataType;
+import org.mule.api.transformer.MessageTransformer;
+import org.mule.api.transformer.Transformer;
+import org.mule.api.transformer.TransformerException;
+import org.mule.api.transformer.TransformerMessagingException;
+import org.mule.api.transport.PropertyScope;
+import org.mule.config.i18n.CoreMessages;
+import org.mule.transformer.TransformerUtils;
+import org.mule.transformer.types.DataTypeFactory;
+import org.mule.transformer.types.MimeTypes;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class TransformationService
+{
+
+    private static final Log logger = LogFactory.getLog(TransformationService.class);
+
+    private MuleContext muleContext;
+
+    public TransformationService(MuleContext muleContext)
+    {
+        this.muleContext = muleContext;
+    }
+
+    /**
+     * Will apply a list of transformers to the payload of the message. This *Will* change the payload of the
+     * message. This method provides the only way to alter the paylaod of this message without recreating a
+     * copy of the message
+     * @param event the event being processed
+     * @param transformers the transformers to apply to the message payload
+     * @throws TransformerException if a transformation error occurs or one or more of the transformers passed in a
+     * are incompatible with the message payload
+     */
+    public void applyTransformers(MuleMessage message, MuleEvent event, List<? extends Transformer> transformers) throws MuleException
+    {
+        applyTransformers(message, event, transformers, null);
+    }
+
+    /**
+     * Will apply a list of transformers to the payload of the message. This *Will* change the payload of the
+     * message. This method provides the only way to alter the paylaod of this message without recreating a
```

Comment:
"This method provides the only way to alter the paylaod" this is clearly not true at thre is at least another method that does the same